### PR TITLE
PMM-10994 New integration setup

### PR DESCRIPTION
**Pull request by peterSirotnak**

### CODE REVIEW

File: .github/workflows/pmm2-integration-tests.yml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      PS_VERSION: ${{ github.event.inputs.ps_version }}
+      MONGO_VERSION: ${{ github.event.inputs.mongo_version }}
+      MONGO_SETUP: ${{ github.event.inputs.mongo_setup }}
+      
+      
+
+    steps:
+      - name: Tests of pmm server version ${{ github.event.inputs.pmm_server_version }} for integration flag ${{ github.event.inputs.integration_setup }}
+        if: ${{ github.event_name == 'workflow_dispatch' && env.SHA != 'null' }}
+        uses: percona-platform/github-status-action@v1
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply .github/workflows/pmm2-integration-tests.yml]: Okay, changed.

File: .github/workflows/pmm2-integration-tests.yml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        with:
+          context: 'pmm2-ui-tests'
+          description: 'Tests execution has been started'
+          state: 'pending'
+          repository: ${{ github.event.inputs.repo }}
+          target_url: 'https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}'
+          sha: ${{ env.SHA }}
+
+      - name: Install NodeJS v16
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply .github/workflows/pmm2-integration-tests.yml:83]: Well our pmm-ui-tests do require specific vesrion of node js. Which is different from one provided on ubuntu-latest

File: .github/workflows/pmm2-integration-tests.yml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        uses: percona-platform/checkout@v2
+        with:
+          ref: ${{ env.UI_TESTS_BRANCH }}
+          repository: percona/pmm-ui-tests
+          path: ./pmm-ui-tests
+
+      - name: Checkout pmm-qa Repo
+        uses: percona-platform/checkout@v2
+        with:
+          # token: ${{ secrets.ROBOT_TOKEN }}
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply .github/workflows/pmm2-integration-tests.yml]: Comment removed.

File: .github/workflows/pmm2-integration-tests.yml [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+          docker network create pmm-integration-network
+          docker run -d --restart always -e PERCONA_TEST_PLATFORM_ADDRESS=https://check-dev.percona.com:443 --network="pmm-integration-network" --publish 80:80 --publish 443:443 --name pmm-integration-server perconalab/pmm-server:${{ env.PMM_SERVER_VERSION }}
+
+      - name: Create env file
+        run: |
+          cd pmm-qa/pmm-integration/
+          touch .env
+          echo CI=true >> .env
+
+      - name: Setup Docker client for PMM-Server
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply .github/workflows/pmm2-integration-tests.yml:113]: Not yet but might be used for different setups, for example external service setup.

File: .github/workflows/pmm2-integration-tests.yml [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+          ref: ${{ env.QA_TESTS_BRANCH }}
+
+      - name: Start PMM-Sever
+        run: |
+          docker network create pmm-integration-network
+          docker run -d --restart always -e PERCONA_TEST_PLATFORM_ADDRESS=https://check-dev.percona.com:443 --network="pmm-integration-network" --publish 80:80 --publish 443:443 --name pmm-integration-server perconalab/pmm-server:${{ env.PMM_SERVER_VERSION }}
+
+      - name: Create env file
+        run: |
+          cd pmm-qa/pmm-integration/
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply .github/workflows/pmm2-integration-tests.yml]: Changed

File: .github/workflows/pmm2-integration-tests.yml [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        run: |
+          sudo docker ps -a
+          printenv INTEGRATION_FLAG
+          cd ./pmm-ui-tests
+          npm install
+          ./node_modules/.bin/codeceptjs run --debug --steps --reporter mochawesome -c pr.codecept.js  --grep '${{ env.INTEGRATION_FLAG }}'
+
+      - name: Generating the report
+        if: ${{ always() && (steps.pmm-ui-tests-all.outcome != 'skipped' || steps.pmm-ui-tests-tagged.outcome != 'skipped' ) }}
+        run: |
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply .github/workflows/pmm2-integration-tests.yml:136]: But it is failing right now on missing ROBOT TOKEN in secrets, do we want to not fail pipeline of failure?

File: pmm-integration/helpers/validateArgs.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import { exit } from "process";
+import availableCommandsLineArgs, { availableArgsMap } from "../availableArgs";
+
+const validateArgs = async (args: string[]) => {
+    args.forEach((arg) => {
+        if(availableCommandsLineArgs.findIndex(option => arg.includes(option)) < 0) {
```
[the remark this answers is not in the document]
Comment (peterSirotnak) [reply pmm-integration/helpers/validateArgs.ts:6]: This one iterates trough all provided args from the command line and validates them. if provided argument is not present in the list of availible args (findIndes =-1) Then it stops the program and prints out all available args.

**Review by puneet0191 — APPROVED** [review]
Looks good to me, some comments. 

1) Add empty last lines where needed. 
2) Add how to use documentation, like a readme 
3) We should add support for `--addclient=ps,1` `--addclient=pdpgsql,1` --addclient=modb,1`

Point number 3 will save us a lot of time in refactoring other dependent jobs.

**Pull request merged**